app/plugin.go

```diff
@@ -429,10 +440,25 @@ func (a *App) ShutDownPlugins() {
 	a.PluginEnv = nil
 }
 
+func getKeyHash(key string) (string, *model.AppError) {
+	if len(key) == 0 {
+		return "", model.NewAppError("getKeyHash", "app.plugin.key_value.key.app_error", nil, "", http.StatusBadRequest)
+	}
```

Comment:
Hashing an empty string is perfectly valid (`e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`). It'd be a little weird, but probably okay if we allowed it for keys. Would at least simplify the code in this PR. 0/5 though.